node/network/protocol/src/request_response.rs

```diff
@@ -0,0 +1,109 @@
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Overview over request/responses as used in `Polkadot`.
+//!
+//! enum  Protocol .... List of all supported protocols.
+//!
+//! enum  Requests  .... List of all supported requests, each entry matches one in protocols, but
+//! has the actual request as payload.
+//!
+//! struct IncomingRequest .... wrapper for incoming requests, containing a sender for sending
+//! responses.
+//!
+//! struct OutgoingRequest .... wrapper for outgoing requests, containing a sender used by the
+//! networking code for delivering responses/delivery errors.
+//!
+//! trait `IsRequest` .... A trait describing a particular request. It is used for gathering meta
+//! data, like what is the corresponding response type.
+//!
+//!  Versioned (v1 module): The actual requests and responses as sent over the network.
+
+use std::borrow::Cow;
+use std::time::Duration;
+
+use futures::channel::mpsc;
+use strum::EnumIter;
+
+pub use sc_network::config as network;
+pub use sc_network::config::RequestResponseConfig;
+
+/// All requests that can be sent to the network bridge.
+pub mod request;
+pub use request::{IncomingRequest, OutgoingRequest, Requests};
+
+///// Multiplexer for incoming requests.
+// pub mod multiplexer;
+
+/// Actual versioned requests and responses, that are sent over the wire.
+pub mod v1;
+
+/// A protocol per subsystem seems to make the most sense, this way we don't need any dispatching
+/// within protocols.
+#[derive(Copy, Clone, Debug, Hash, PartialEq, Eq, EnumIter)]
+pub enum Protocol {
+	/// Protocol for availability fetching, used by availability distribution.
+	AvailabilityFetching,
+}
+
+impl Protocol {
+	/// Get a configuration for a given Request response protocol.
+	///
+	/// Returns a receiver for messages received on this protocol and the requested
+	/// `ProtocolConfig`.
+	///
+	/// See also `dispatcher::RequestDispatcher`,  which makes use of this function and provides a more
+	/// high-level interface.
+	pub fn get_config(
+		self,
+	) -> (
+		mpsc::Receiver<network::IncomingRequest>,
+		RequestResponseConfig,
+	) {
+		let p_name = self.into_protocol_name();
+		// Channel size again a number I made up. Hundreds of validators will start requesting
+		// their chunks once they see a candidate awaiting availability on chain. Given that they
+		// will see that block at different times (due to network delays), 100 seems big enough to
+		// accomodate for "bursts", assuming we can service requests relatively quickly, which
+		// would need to be measured as well.
+		let (tx, rx) = mpsc::channel(100);
```

Comment:
Extract this constant.

Reply:
I made it a function, that spits out values depending on the protocol.